app/controllers/watchers_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class WatchersController < ApplicationController
+  load_and_authorize_resource :issue
+  load_and_authorize_resource through: :issue
+
+  # POST /issues/:issue_id/watchers
+  def create
+    if @watcher.save
+      flash.now[:notice] = t('watchers.creation.success')
+
+      # send email to the watching user
+      @watcher.inform_started_watching
+    else
+      flash.now[:error] = t('watchers.creation.failure')
+      @watcher = nil
+    end
+
+    respond_to do |format|
+      format.js { render 'update' }
+    end
+  end
+
+  # DELETE /issues/:issue_id/watchers/:id
+  def destroy
+    if @watcher.destroy
+      flash.now[:notice] = t('watchers.deletion.success')
+      @watcher.inform_stopped_watching
+      @watcher = nil
```

Comment:
why nil?

Reply:
In the view I am checking if the watcher is nil

Reply:
check if watcher.persisted? that would make more sense.